pypdf/generic/_base.py

```diff
@@ -282,9 +282,27 @@ def indirect_reference(self) -> "IndirectObject":  # type: ignore[override]
 
     def get_object(self) -> Optional["PdfObject"]:
         obj = self.pdf.get_object(self)
-        if obj is None:
-            return None
-        return obj.get_object()
+        return obj
```

Comment:
No need to assign the variable in this case - couldn't we just return the resolved object?